pkg/safemath/safemath.go

```diff
@@ -0,0 +1,33 @@
+package safemath
+
+import "math/bits"
+
+// This is a constant-time version of the following function:
```

Comment:
I would add a description at the beginning. Something along the lines of: "this takes a uint64 to an int16 and outputs their addition as well as the ocurrence of an overflow or underflow"

Reply:
So readers wouldn't need to dive into the explanation to get some context. Very cool explanation nonetheless it makes all the bit handling makes sense!